desktop_version/src/Music.cpp

```diff
@@ -92,81 +131,263 @@ class SoundTrack
+                }
+                if (faudio_buffer.PlayLength == 0) return;
+            } else {
+                return;
+            }
+        }
+        t->sample_pos += faudio_buffer.PlayLength;
+        FAudioSourceVoice_SubmitSourceBuffer(musicVoice, &faudio_buffer, NULL);
+    }
+    static void swapBuffers(FAudioVoiceCallback* callback, void* ctx) {
+        MusicTrack* t = (MusicTrack*)ctx;
+        std::swap(t->decoded_buf_playing, t->decoded_buf_reserve);
+    }
+
+    //Lifted from sdl_mixer
+    static void parseComments(MusicTrack* t, char** comments, int comment_list_length) {
+        int loopend = 0;
+        for (int i = 0; i < comment_list_length; i++) {
+            char *param = SDL_strdup(comments[i]);
+            char *argument = param;
+            char *value = SDL_strchr(param, '=');
+            if (value == NULL) {
+                value = param + SDL_strlen(param);
+            } else {
+                *(value++) = '\0';
+            }
 
-static std::vector<SoundTrack> soundTracks;
-static std::vector<MusicTrack> musicTracks;
+            /* Want to match LOOP-START, LOOP_START, etc. Remove - or _ from
+            * string if it is present at position 4. */
+            //_Mix_IsLoopTag inlined
+            char buf[5];
+            SDL_strlcpy(buf, argument, 5);
+            if (SDL_strcasecmp(buf, "LOOP") == 0 && ((argument[4] == '_') || (argument[4] == '-'))) {
+                SDL_memmove(argument + 4, argument + 5, SDL_strlen(argument) - 4);
+            }
+
+            if (SDL_strcasecmp(argument, "LOOPSTART") == 0)
+                t->loopbegin = _Mix_ParseTime(value, t->format.nSamplesPerSec);
+            else if (SDL_strcasecmp(argument, "LOOPLENGTH") == 0) {
+                t->looplength = SDL_strtoll(value, NULL, 10);
+            } else if (SDL_strcasecmp(argument, "LOOPEND") == 0) {
+                loopend = _Mix_ParseTime(value, t->format.nSamplesPerSec);
+            }
+            
+            SDL_free(param);
+        }
+        if (loopend != 0) {
+            t->looplength = loopend - t->loopbegin;
+        }
+    }
+    //_Mix_ParseTime
+    static int _Mix_ParseTime(char *time, long samplerate_hz) {
+        char *num_start, *p;
+        Sint64 result;
+        char c;
+        int val;
+
+        /* Time is directly expressed as a sample position */
+        if (SDL_strchr(time, ':') == NULL) {
+            return SDL_strtoll(time, NULL, 10);
+        }
+
+        result = 0;
+        num_start = time;
+
+        for (p = time; *p != '\0'; ++p) {
+            if (*p == '.' || *p == ':') {
+                c = *p; *p = '\0';
+                if ((val = SDL_atoi(num_start)) < 0)
+                    return -1;
+                result = result * 60 + val;
+                num_start = p + 1;
+                *p = c;
+            }
 
-/* End SDL_mixer wrapper */
+            if (*p == '.') {
+                double val_f = SDL_atof(p);
+                if (val_f < 0) return -1;
```

Comment:
I know that this is done elsewhere in the codebase, but I'd really like for this to be a proper if-block instead of a one-liner - e.g. please do this instead:
```c++
if (cond)
{
    statement;
}
```